config_manager.go

```diff
@@ -189,7 +206,7 @@ func (cm *ConfigManager) loadConfigFile() error {
 	file, err := ioutil.ReadFile(cm.Flags.ConfigFile)
 	// don't error if the default config file isn't found
 	if os.IsNotExist(err) && cm.Flags.ConfigFile == DefaultConfigFile {
-		return nil
+		return defaultConfigDoesNotExist
```

Comment:
Completely agree on users not wanting to read code on failures, and they definitely shouldn't have to. However, IMO, we should validate that all required parameters have been supplied and not worry about where they came from. If a user specifies all required parameters via the CLI, a config file isn't necessary and thus we can't justify suggesting something is wrong.

The only situation I can think of where this falls down is when an optional parameter is specified in a config file and r_s2 is never asked to load it. One solution to that might be printing the state of all settings when `-D` is used. I'd prefer to hold off doing that ATM as it hasn't been something that comes up in practice. We can always revisit if things change.

Let me know if I've missed a scenario here or there's a hole in my logic.